ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/DropSourceCommand.java

```diff
@@ -58,14 +67,22 @@ public DdlCommandResult run(MetaStore metaStore, boolean isValidatePhase) {
         new DropTopicCommand(dataSource.getKsqlTopic().getTopicName());
     metaStore.deleteSource(sourceName);
     dropTopicCommand.run(metaStore, isValidatePhase);
-    if (!isValidatePhase && dataSource.getKsqlTopic().getKsqlTopicSerDe().getSerDe()
-                            == DataSource.DataSourceSerDe.AVRO) {
+    if (!isValidatePhase && withTopic) {
       try {
-        schemaRegistryClient
-            .deleteSubject(sourceName + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX);
+        kafkaTopicClient.deleteTopics(Arrays.asList(dataSource.getKsqlTopic().getKafkaTopicName()));
       } catch (Exception e) {
-        throw new KsqlException("Could not clean up the schema registry for topic: " + sourceName,
-                                e);
+        throw new KsqlException("Could not delete the corresponding kafka topic: "
```

Comment:
If this fails maybe we should save the exception and still try and delete from the schema registry?
Maybe we should retry, too? I guess it is hard to tell from the Exception if it is retryable

Reply:
Good point, added retry block for both delete topic and delete schema.